pallets/gear/src/internal.rs

```diff
@@ -586,15 +593,45 @@ where
             // Message is going to be inserted into mailbox.
             //
             // No hold bound checks required, because gas_limit isn't less than threshold.
-            if gas_limit >= threshold {
+            to_mailbox = gas_limit >= threshold;
+            if to_mailbox {
                 // Cutting gas for storing in mailbox.
-                GasHandlerOf::<T>::cut(sender_node, dispatch.id(), gas_limit)
+                let gas_amount = gas_for_delay.saturating_add(gas_limit);
+                GasHandlerOf::<T>::cut(sender_node, dispatch.id(), gas_amount)
                     .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
 
                 // TODO: adapt this line if gasful sending appears for reservations (#1828)
                 if let Some(reservation_id) = reservation {
                     Self::remove_gas_reservation_with_task(dispatch.source(), reservation_id);
                 }
+            } else {
+                GasHandlerOf::<T>::cut(sender_node, dispatch.id(), gas_for_delay)
```

Comment:
If it won't go to mailbox, there should be gas_limit or even zero
```suggestion
                GasHandlerOf::<T>::cut(sender_node, dispatch.id(), gas_limit)
```

Reply:
write test for it pls

Reply:
discussed. No change needed